test/units/galaxy/test_token.py

```diff
@@ -96,3 +190,51 @@ def test_token_from_file_missing(b_token_file):
 @pytest.mark.parametrize('b_token_file', ['file'], indirect=True)
 def test_token_none(b_token_file):
     assert GalaxyToken(token=NoTokenSentinel).get() is None
+
+
+def test_parse_bearer_token(monkeypatch):
+    monkeypatch.setattr(token_file, 'REQUIRED_JWT_HEADER_KEYS', set(['alg']))
+    monkeypatch.setattr(token_file, 'REQUIRED_JWT_HEADER_VALUES', {'alg': ['RS256']})
+
+    # result from https://www.rfc-editor.org/rfc/rfc7515#appendix-A.2.1 to test validating A.2.2
+    token_info = (
+        "eyJhbGciOiJSUzI1NiJ9.eyJpc3MiOiJqb2UiLA0KICJleHAiOjEzMDA4MTkzODAsDQogImh0"
+        "dHA6Ly9leGFtcGxlLmNvbS9pc19yb290Ijp0cnVlfQ.cC4hiUPoj9Eetdgtv3hF80EGrhuB__"
+        "dzERat0XF9g2VtQgr9PJbu3XOiZj5RZmh7AAuHIm4Bh-0Qc_lF5YKt_O8W2Fp5jujGbds9uJd"
+        "bF9CUAr7t1dnZcAcQjbKBYNX4BAynRFdiuB--f_nZLgrnbyTyWzO75vRK5h6xBArLIARNPvkS"
+        "jtQBMHlb1L07Qe7K0GarZRmB_eSN9383LcOLn6_dO--xi12jzDwusC-eOkHWEsqtFZESc6BfI"
+        "7noOPqvhJ1phCnvWh6IeYI2w9QOYEUipUTI8np6LbgGY9Fs98rqVt5AXLIhWkWywlVmtVrBp0"
+        "igcN_IoypGlUPQGe77Rw"
+    )
+
+    e = "AQAB"
+    n = (
+        "ofgWCuLjybRlzo0tZWJjNiuSfb4p4fAkd_wWJcyQoTbji9k0l8W26mPddxHmfHQp-Vaw-4qPC"
+        "JrcS2mJPMEzP1Pt0Bm4d4QlL-yRT-SFd2lZS-pCgNMsD1W_YpRPEwOWvG6b32690r2jZ47soM"
+        "Zo9wGzjb_7OMg0LOL-bSf63kpaSHSXndS5z5rexMdbBYUsLA9e-KXBdQOS-UTo7WTBEMa2R2C"
+        "apHg665xsmtdVMTBQY4uDZlxvb3qCo5ZwKh9kG4LT6_I5IhlJH7aGhyxXFvUK-DWNmoudF8NA"
+        "co9_h9iaGNj8q2ethFkMLs91kzk2PAcDTW9gb54h4FRWyuXpoQ"
+    )
+    public_numbers = token_file.PublicNumbers(n=n, e=e)
+    token_file.parse_bearer_token(token_info, public_numbers=public_numbers)
+
+    with pytest.raises(token_file.InvalidSignature):
```

Comment:
It is the best practice to always use `pytest.raises()` with the `match=` argument. Otherwise, expect the future false-positives no go undetected.

Reply:
There is no message but I didn't think our tests should be asserting that (if cryptography adds a message, our tests would still be valid). I checked the cryptography unit tests, and they use this pattern without `match`.

Reply:
Yes, of course.